cpu/cpustress.py

```diff
@@ -0,0 +1,307 @@
+            for cpus in range(1, totalcpus):
+                self.log.info("cpu%s going offline" % cpus)
+                cpu.offline(cpus)
+                time.sleep(timeout)
+            self.log.info("Online CPU's in reverse order %s", totalcpus)
+            for cpus in range(totalcpus, -1, -1):
+                self.log.info("cpu%s going online" % cpus)
+                cpu.online(cpus)
+                time.sleep(timeout)
+            self.log.info("Offline CPU's in reverse order %s", totalcpus)
+            for cpus in range(totalcpus, -1, -2):
+                self.log.info("cpu%s going offline" % cpus)
+                cpu.offline(cpus)
+                time.sleep(timeout)
+            self.log.info("Online CPU's in serial")
+            for cpus in range(0, totalcpus):
+                self.log.info("cpu%s going online" % cpus)
+                cpu.online(cpus)
+                time.sleep(timeout)
+
+    def single_cpu_toggle(self, iteration, timeout):
+        """
+        Offline-online single cpu for given iteration
+        and loop over all cpus.
+        @BUG: https://lkml.org/lkml/2017/6/12/212
+        """
+        for cpus in range(1, totalcpus):
+            for _ in range(iteration):
+                self.log.info("cpu%s going offline" % cpus)
+                cpu.offline(cpus)
+                time.sleep(timeout)
+                self.log.info("cpu%s going online" % cpus)
+                cpu.online(cpus)
+
+    def cpu_toggle_one_by_one(self, iteration, timeout):
+        """
+        Wait for the given timeout between Off/On single cpu.
+        loop over all cpus for given iteration.
+        """
+        for _ in range(iteration):
+            for cpus in range(totalcpus):
+                self.log.info("cpu%s going offline" % cpus)
+                cpu.offline(cpus)
+                time.sleep(timeout)
+                self.log.info("cpu%s going online" % cpus)
+                cpu.online(cpus)
+
+    def multiple_cpus_toggle(self, iteration, timeout):
+        """
+        offline/online multiple CPUS at once
+        enable different smt options to offline multiple cpus
+        @BUG : https://lkml.org/lkml/2017/7/3/21
+        """
+        if self.__isSMT():
+            for _ in range(iteration):
+                self.log.info("SMT toggle")
+                process.run("ppc64_cpu --smt=off && ppc64_cpu --smt=on",
+                            shell=True)
+            time.sleep(timeout)
+        else:
+            self.log.info('Machine is not SMT capable')
+
+    def pinned_cpu_stress(self, iteration, timeout):
+        """
+        Set process affinity and do cpu off on
+        @BUG : https://lkml.org/lkml/2017/5/30/122
+        """
+        nodes = []
+        self.log.info("\nCreate %s pids and set proc affinity", totalcpus)
+        for proc in range(0, totalcpus):
+            obj = process.SubProcess("while :; do :; done &", shell=True)
+            pid = obj.start()
```

Comment:
i think it can be 
`pids.append(process.SubProcess("while :; do :; done &", shell=True).start())`